fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -2574,8 +2574,10 @@ public void addFrontend(FrontendNodeType role, String host, int editLogPort) thr
             fe = new Frontend(role, nodeName, host, editLogPort);
             frontends.put(nodeName, fe);
             if (role == FrontendNodeType.FOLLOWER || role == FrontendNodeType.REPLICA) {
-                ((BDBHA) getHaProtocol()).addHelperSocket(host, editLogPort);
+                BDBHA ha = (BDBHA) haProtocol;
+                ha.addHelperSocket(host, editLogPort);
                 helperNodes.add(Pair.create(host, editLogPort));
+                ha.addUnstableNode(host, getFollowerCnt());
```

Comment:
If the master transferred between the addFrontend and dropBackend, what will happen?

Reply:
If master transferred during the join progress, the new master won't have the unstable node info. But that is ok because transferring master means there is quorum size node alive, there is no need to wait the joining node's ack before completing a write operation.